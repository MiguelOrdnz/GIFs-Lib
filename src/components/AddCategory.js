import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {
    
    const [inputValue, setInputValue] = useState('');

    const handleOnChange = (e) =>{
        setInputValue(e.target.value);
    } 

    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 0){
            setCategories( (categories) => [...new Set([inputValue, ...categories])] )
            setInputValue('')
        }

    }

    return (
        <form onSubmit = { handleSubmit }>
            <input 
                placeholder='Buscar'
                type='text' 
                value = { inputValue }
                onChange = { handleOnChange }
            />
        </form>
    )
}

AddCategory.propTypes= {
    setCategories: PropTypes.func.isRequired
}