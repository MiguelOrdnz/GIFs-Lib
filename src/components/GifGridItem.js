import React from 'react'
import PropTypes from 'prop-types'


export const GifGridItem = (props) => {
    const {id, title, url} = props
    return (
        <div className="GifGridItem animate__animated animate__fadeIn animate__delay-1s">
            <img id={id} src={url} alt={title}/>
            <p>{title}</p> 
        </div>
    )
}

GifGridItem.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
