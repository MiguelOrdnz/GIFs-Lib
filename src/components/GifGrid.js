import React from 'react'
import { useFetchGiphy } from '../hooks/useFetchGiphy'
import { GifGridItem } from './GifGridItem'

export const GifGrid = ({ category }) => {    

    const {imgs, loading} = useFetchGiphy(category)

    return (
        <>
            <h3>{ category }</h3>
            <hr/>
            { loading && 'loading..'}

            <div className='GifGrid'>
                {
                    imgs.map( gif => <GifGridItem key={gif.id} {...gif}/> )
                }
            </div>
        </>
    )
}
