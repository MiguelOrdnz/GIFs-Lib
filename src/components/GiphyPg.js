import React, { useState } from 'react'
import { AddCategory } from "./AddCategory";
import { GifGrid } from "./GifGrid";

export const GiphyPg = () => {
    const [categories, setCategories] = useState([])

    return (
        <div>
            <h1>Giphy Playground</h1>
            <AddCategory setCategories = { setCategories }/>
            <div>
            {
                categories.map(( category ) => (
                    <GifGrid 
                    key={ category }
                    category={ category }/>
                )
                )
            }
            </div>
        </div>
    )
}
