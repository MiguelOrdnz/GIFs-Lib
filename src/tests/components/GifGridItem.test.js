import React from 'react'
import { shallow} from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'

describe('Testint <GifGridItem />>', () => {
    const id = ''
    const title = ''
    const url = ''
    const wrapper = shallow(<GifGridItem id = {id} title = {title} url = {url}/>)

    test('Match Snapshot', () => {
        expect( wrapper ).toMatchSnapshot();
    })
    
    it('Contain a paragrah with the title', () => {
        const p = wrapper.find('p')
        expect(wrapper.text().trim()).toBe(title)
    });
    
});
