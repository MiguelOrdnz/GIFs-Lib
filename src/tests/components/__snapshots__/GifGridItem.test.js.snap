// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testint <GifGridItem />> Match Snapshot 1`] = `
<div
  className="GifGridItem animate__animated animate__fadeIn animate__delay-1s"
>
  <img
    alt=""
    id=""
    src=""
  />
  <p />
</div>
`;
